tests/bal_business_logic_suite.py

```diff
@@ -380,12 +380,12 @@ class Test_hasCapability_default(FixtureAugmentedTestCase):
     def test_when_hasCapability_called_then_expected_capabilities_reported(self):
         self.assertFalse(self._manager.hasCapability(Manager.Capability.kStatefulContexts))
         self.assertFalse(self._manager.hasCapability(Manager.Capability.kCustomTerminology))
-        self.assertFalse(self._manager.hasCapability(Manager.Capability.kDefaultEntityReferences))
```

Comment:
This deletion puzzled me, as it suggests `Manager.Capability.kDefaultEntityReferences` existed already prior to this PR.

Reply:
I'm not super sure what you mean, perhaps there is confusion on the distinction between advertised capabilities, and the API methods associated with those capabilities?

I.e. now that the `defaultEntityReference` API method is supported, BAL can advertise that the associated capability `kDefaultEntityReferences` is supported.  

Before this PR the capability was not supported, so we had to `assertFalse`, whereas now (further down in this test) we `assertTrue`.

Reply:
Added that particular missing type hint in 02c21f24.  

Looking through the other changes, the other opportunity for type hints is the main `BasicAssetLibrary.py` and the `bal_business_logic_suite.py` tests, neither of which currently make use of type annotations, even though they could.

So I went with laziness/consistency and didn't add annotations there. I'll add an issue though,

Reply:
Added #123 and put "good first issue" / "help wanted" labels on it :wink: 

The "help wanted" label should mean it'll appear on https://clotributor.dev/ - though perhaps that's for the main repo only :thinking:

Reply:
I think it's likely more lack of understanding of what is going on in the library more than anything.

For example `test_when_no_default_then_entity_ref_is_None`  is pretty self explanatory.

Although I've just looked again, and I think it's clicked that

`lambda idx, value: operator.setitem(results, idx, value),`

is the success callback?

Reply:
Yeah that's right. Since we don't (yet) have the convenience signatures (https://github.com/OpenAssetIO/OpenAssetIO/pull/1407) we have to use the clunky callback-based API.